src/clusterfuzz/_internal/system/archive.py

```diff
@@ -34,111 +36,313 @@
     ZIP_FILE_EXTENSIONS + TAR_FILE_EXTENSIONS + LZMA_FILE_EXTENSIONS)
 
 
-class ArchiveType:
-  """Type of the archive."""
-  UNKNOWN = 0
-  ZIP = 1
-  TAR = 2
-  TAR_LZMA = 3
+def _is_attempting_path_traversal(archive_name, output_dir, filename) -> bool:
+  """Detects whether there is a path traversal attempt.
 
+  Args:
+      archive_name (str): the name of the archive.
+      output_dir (str): the output directory.
+      filename (str): the name of the file being checked.
 
-class ArchiveFile:
-  """File in an archive."""
+  Returns:
+      bool: Whether there is a path traversal attempt
+  """
+  absolute_file_path = os.path.join(output_dir, os.path.normpath(filename))
```

Comment:
can we do `output_dir = os.path.realpath(output_dir)` before this?

If `output_dir`  has `..` in it, e.g. `../a` or some other variation (which is trusted), then

```
absolute_file_path == '../a/filename'
real_file_path = '<some other dir>/a/filename`
```

Then this will fail, even though there's no directory traversal.

Reply:
Oh, nice, good point, I didn't think about this!